src/libraries/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs

```diff
@@ -1960,6 +1960,42 @@ public static void Xml_TypeWithSpecialCharacterInStringMember()
         Assert.Equal(x.Name, y.Name);
     }
 
+    [Fact]
+    public static void Xml_TypeInCollectibleALC()
+    {
+        WeakReference weakRef;
+
+#if !XMLSERIALIZERGENERATORTESTS
+        ExecuteAndUnload("System.Collections.Specialized", "System.Collections.Specialized.StringCollection", out weakRef);
+#else
```

Comment:
Are you sure this is getting loaded into the collectible ALC? I remember something about types from .NET itself will get loaded into the default ALC regardless unless you provide a custom assembly resolver. Can you verify that the loaded assembly is in your collectible ALC?